### PR TITLE
tools/pd-ctl/pdctl: pd-ctl interact mode autocomplete

### PR DESCRIPTION
**Pull request by jiashiwen**
### What problem does this PR solve?

https://github.com/pingcap/pd/pull/2299

### What is changed and how it works?
interactive mode ,use "tab" to auto complete command and flags

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

run pd-ctl as interactive mode
press 'tab' show command will be used

Code changes

add new varable 
readlinecompleter *readline.PrefixCompleter

add new func
func GenCompleter(cmd *cobra.Command) []readline.PrefixCompleterInterface {
	pc := []readline.PrefixCompleterInterface{}
	if len(cmd.Commands()) != 0 {
		for _, v := range cmd.Commands() {
			//fmt.Println(strings.Split(v.Use, " ")[0]
			if v.HasFlags() {
				flagspc := []readline.PrefixCompleterInterface{}

				flaguseages := strings.Split(strings.Trim(v.Flags().FlagUsages(), " "), "\n")

				for i := 0; i < len(flaguseages)-1; i++ {
					flagspc = append(flagspc, readline.PcItem(strings.Split(strings.Trim(flaguseages[i], " "), " ")[0]))
				}
				flagspc = append(flagspc, GenCompleter(v)...)
				pc = append(pc, readline.PcItem(strings.Split(v.Use, " ")[0], flagspc...))

			} else {
				pc = append(pc, readline.PcItem(strings.Split(v.Use, " ")[0], GenCompleter(v)...))
			}
		}
	}
	return pc
}

func getMainCmd add new expression
	readlinecompleter = readline.NewPrefixCompleter(GenCompleter(rootCmd)...)

func loop() add  
AutoComplete:      readlinecompleter,

- Has configuration change
- Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/pingcap/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tidb-ansible`](https://github.com/pingcap/tidb-ansible):
- Need to cherry-pick to the release branch

### CODE REVIEW

File: tools/pd-ctl/pdctl/ctl.go [hunk 1]
```diff
@@ -190,3 +194,27 @@ func loop() {
 		Start(args)
 	}
 }
+
+func genCompleter(cmd *cobra.Command) []readline.PrefixCompleterInterface {
+	pc := []readline.PrefixCompleterInterface{}
+	if len(cmd.Commands()) != 0 {
+		for _, v := range cmd.Commands() {
+			//fmt.Println(strings.Split(v.Use, " ")[0]
```
Comment (Yisaer) [inline tools/pd-ctl/pdctl/ctl.go:202]: please remove the debug code.

File: tools/pd-ctl/pdctl/ctl.go [hunk 2]
```diff
@@ -190,3 +194,27 @@ func loop() {
 		Start(args)
 	}
 }
+
+func genCompleter(cmd *cobra.Command) []readline.PrefixCompleterInterface {
```
Comment (Yisaer) [inline tools/pd-ctl/pdctl/ctl.go:198]: Could you add a unit test for this function?

**Comment by jiashiwen** [comment]
how to add sign-off ?

**Comment by JmPotato** [comment]
> how to add sign-off ?

You can get some guidances from [here](https://docs.pi-hole.net/guides/github/how-to-signoff)!

**Comment by jiashiwen** [comment]
> > how to add sign-off ?
> 
> You can get some guidances from [here](https://docs.pi-hole.net/guides/github/how-to-signoff)!

I have sign-off my commit but DCO not pass

**Comment by JmPotato** [comment]
> > > how to add sign-off ?
> > 
> > 
> > You can get some guidances from [here](https://docs.pi-hole.net/guides/github/how-to-signoff)!
> 
> I have sign-off my commit but DCO not pass

A signed off commit should look like this:

image

Maybe you can try this [script](https://gist.github.com/disksing/4122718e910e33138032a72a45688480) to sign off all your commits?

**Comment by jiashiwen** [comment]
> > > > how to add sign-off ?
> > > 
> > > 
> > > You can get some guidances from [here](https://docs.pi-hole.net/guides/github/how-to-signoff)!
> > 
> > 
> > I have sign-off my commit but DCO not pass
> 
> A signed off commit should look like this:
> 
> image
> 
> Maybe you can try this [script](https://gist.github.com/disksing/4122718e910e33138032a72a45688480) to sign off all your commits?

I forck the project from pingcap.Perhaps I need forck from tikv an repeat pr?

**Comment by JmPotato** [comment]
@jiashiwen There are indeed some problems about the sign-off because of yesterday's transferring. Maybe re-forking the tikv/pd is a good idea. Just remember to sign off your new commits after!

**Comment by jiashiwen** [comment]
I have pr again

**Pull request closed without merging**